### PR TITLE
Changelog generation

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

Adding changelogs, man pages, release notes to pipelines. All changes that are needed for a release are in a diff which we can re-use in later pipeline steps.

### What issues does this PR fix or reference?
Fixes:

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by twangboy — DISMISSED** [review]

File: tools/changelog.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
+REPO_ROOT = pathlib.Path(__file__).resolve().parent.parent
+
+# Define the command group
+cl = command_group(name="changelog", help="Changelog tools", description=__doc__)
+
+
+def changelog(version):
+    """
+    Return a the full changelog generated by towncrier.
```
Comment (Ch3LL) [inline tools/changelog.py:26]:
```suggestion
    Return the full changelog generated by towncrier.
```

**Review by Ch3LL — CHANGES_REQUESTED** [review]
I also have a question. I don't think this is pushing a PR with the changelog and release notes. How are we going to ensure these updates to the changelog and release notes will be reflected in  the master branch?

File: tools/changelog.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    orig = ctx.run(
+        "sed",
+        f"s/Version: .*/Version: {salt_version}/g",
+        "pkg/rpm/salt.spec",
+        capture=True,
+        check=True,
+    ).stdout.decode()
+    dt = datetime.datetime.utcnow()
+    date = dt.strftime("%a %b %d %Y")
+    header = f"* {date} SaltStack Packaging Team <packaging@saltstack.com> - {salt_version}\n"
```
Comment (dmurphy18) [inline tools/changelog.py:137]: This needs to be updated to the latest:
```
Salt Project Packaging <saltproject-packaging@vmware.com>
```

File: tools/changelog.py [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+    changes = pkg_changelog(salt_version)
+    formated = "\n".join([f"  {_.replace('-', '*', 1)}" for _ in changes.split("\n")])
+    dt = datetime.datetime.utcnow()
+    date = dt.strftime("%a, %d %b %Y %H:%M:%S +0000")
+    tmpchanges = "pkg/rpm/salt.spec.1"
+    with open(tmpchanges, "w") as wfp:
+        wfp.write(f"salt ({salt_version}) stable; urgency=medium\n\n")
+        wfp.write(formated)
+        wfp.write(
+            f"\n -- SaltStack Packaging Team <packaging@saltstack.com> {date}\n\n"
```
Comment (dmurphy18) [inline tools/changelog.py:187]: This needs to be updated to the latest:
```
Salt Project Packaging <saltproject-packaging@vmware.com>
```

**Review by dmurphy18 — CHANGES_REQUESTED** [review]
Being guilty myself of over-using the old 'SaltStack Packaging Team <packaging@saltstack.com> ' and used to have to ask Bryce all the time what the new one was once VMware brought us, but searched and found it.

With the new relenv , should start out correctly and use the correct
```
Salt Project Packaging <saltproject-packaging@vmware.com>
```

File: tools/changelog.py (hunk 1 above)
Comment (s0undt3ch) [reply tools/changelog.py]: Fixed

Comment (s0undt3ch) [reply tools/changelog.py]: > How are we going to ensure these updates to the changelog and release notes will be reflected in the master branch?

When we stage a release, we generate a diff file(like we do for regular PR and branch builds).
When we actually release, the release pipeline will apply the diff we make, commit, tag the version, and push everything.

File: tools/changelog.py (hunk 2 above)
Comment (s0undt3ch) [reply tools/changelog.py]: Fixed

File: tools/changelog.py (hunk 3 above)
Comment (s0undt3ch) [reply tools/changelog.py]: Fixed

**Review by dmurphy18 — APPROVED** [review]

**Pull request merged**